recipe/2025-09-container.md

```diff
@@ -0,0 +1,836 @@
+
+**Service for Network Access:**
+```bicep
+resource service 'core/Service@v1' = if (length(containerSpec.ports ?? []) > 0) {
+  metadata: {
+    name: containerSpec.name
+    namespace: targetNamespace
+    labels: finalLabels
+  }
+  spec: {
+    selector: commonLabels
+    ports: [for port in containerSpec.ports: {
+      name: port.name ?? 'port-${port.port}'
+      port: port.port
+      targetPort: port.targetPort ?? port.port
+      protocol: port.protocol ?? 'TCP'
+    }]
+    type: containerSpec.networking?.expose ?? false ? 'LoadBalancer' : 'ClusterIP'
+  }
+}
+```
+
+#### Resource Creation Logic
+
+The recipe implements the same logic as the current Go renderer:
+
+**Deployment Creation:**
+```bicep
+// Generate deployment with identical metadata and specification
+resource deployment 'apps/Deployment@v1' = {
+  metadata: {
+    name: context.resource.name
+    namespace: environment.namespace
+    labels: {
+      'app.kubernetes.io/name': context.resource.name
+      'app.kubernetes.io/part-of': application.name
+      'app.kubernetes.io/managed-by': 'radius'
+    }
+  }
+  spec: {
+    selector: {
+      matchLabels: {
+        app: application.name
+        resource: context.resource.name
+      }
+    }
+    template: {
+      metadata: {
+        labels: {
+          app: application.name
+          resource: context.resource.name
+          'azure.workload.identity/use': identityRequired ? 'true' : null
+        }
+      }
+      spec: {
+        serviceAccountName: identityRequired ? context.resource.name : null
+        enableServiceLinks: false
+        containers: [
+          {
+            name: context.resource.name
+            image: container.image
+            ports: [for port in items(container.ports): {
+              containerPort: port.value.containerPort
+              protocol: 'TCP'
+            }]
+            env: concat(
+              // Direct environment variables
+              [for env in items(container.env): {
+                name: env.key
+                value: env.value.value
+              } if env.value.value != null],
+              // Secret-referenced environment variables  
+              [for env in items(container.env): {
+                name: env.key
+                valueFrom: {
+                  secretKeyRef: {
+                    name: context.resource.name
+                    key: env.key
+                  }
+                }
+              } if env.value.valueFrom != null]
+            )
+            volumeMounts: [for volume in items(volumes): {
+              name: volume.key
+              mountPath: volume.value.mountPath
+            }]
+            readinessProbe: container.readinessProbe
+            livenessProbe: container.livenessProbe
+          }
+        ]
+        volumes: concat(
+          // Ephemeral volumes
+          [for volume in items(volumes): {
+            name: volume.key
+            emptyDir: {}
+          } if volume.value.kind == 'ephemeral'],
```

Comment:
this is outdated - in the container resource schema we decided to follow the `emptyDir` convention for ephemeral volumes: https://github.com/radius-project/resource-types-contrib/tree/main/Compute/containers

Reply:
> this is outdated - in the container resource schema we decided to follow the `emptyDir` convention for ephemeral volumes: https://github.com/radius-project/resource-types-contrib/tree/main/Compute/containers

I updated the document to follow the `emptyDir` convention.